src/main/java/io/bitrise/trace/step/InjectTraceTask.java

```diff
@@ -254,8 +256,18 @@ static boolean updateBuildScriptContent(final String path) throws IOException {
         final String regex = "buildscript[ \\t\\n\\r]*\\{";
         final Pattern pattern = Pattern.compile(regex);
         final Matcher matcher = pattern.matcher(codeContent);
+        final List<Range> literalsPos = findStringLiterals(codeContent);
```

Comment:
The idea is that we gather the position of String literals from the given text, then when we re searching for the text "buildscript {" we simply jump to the next, if the found one was in a String literal.